csep/core/binomial_evaluations.py

```diff
@@ -102,6 +102,29 @@ def binary_joint_log_likelihood_ndarray(forecast, catalog):
     return sum(first_term.data + second_term.data)
     
     
+def _simulate_catalog(num_events, sampling_weights, sim_fore, random_numbers=None):
+
+    # generate uniformly distributed random numbers in [0,1), this
+    if random_numbers is None:
+        random_numbers = numpy.random.rand(num_events)
+    else:
+        # TODO: ensure that random numbers are all between 0 and 1.
+        pass
+
+    # reset simulation array to zero, but don't reallocate
+    sim_fore.fill(0)
```

Comment:
@bayonato89 @khawajasim can we remove this part of the function? the vector `random_numbers` is not used for anything it seems. the simulation below allocates a new array called `random_num`.